tractor/_portal.py

```diff
@@ -324,23 +328,25 @@ async def open_portal(
     assert actor
     was_connected = False
 
-    async with maybe_open_nursery(nursery) as nursery:
+    async with maybe_open_nursery(nursery, shield=shield) as nursery:
         if not channel.connected():
             await channel.connect()
             was_connected = True
 
         if channel.uid is None:
             await actor._do_handshake(channel)
 
-        msg_loop_cs: trio.CancelScope = await nursery.start(
-            partial(
-                actor._process_messages,
-                channel,
-                # if the local task is cancelled we want to keep
-                # the msg loop running until our block ends
-                shield=True,
+        msg_loop_cs = None
+        if start_msg_loop:
+            msg_loop_cs: trio.CancelScope = await nursery.start(
+                partial(
+                    actor._process_messages,
+                    channel,
+                    # if the local task is cancelled we want to keep
```

Comment:
This is new to shut down logic: we always keep the parent channel message loop running until the very end.